token/core/common/validator.go

```diff
@@ -205,30 +209,43 @@ func (v *Validator[P, T, TA, IA, DS]) verifyIssue(tr IA, ledger driver.Ledger, s
 		}
 		counter += c
 	}
-	if len(tr.GetMetadata()) != counter {
-		return errors.Errorf("more metadata than those validated [%d]!=[%d], [%v]!=[%v]", len(tr.GetMetadata()), counter, tr.GetMetadata(), context.MetadataCounter)
+	if len(action.GetMetadata()) != counter {
+		return errors.Errorf("more metadata than those validated [%d]!=[%d], [%v]!=[%v]", len(action.GetMetadata()), counter, action.GetMetadata(), context.MetadataCounter)
 	}
 
 	return nil
 }
 
-func (v *Validator[P, T, TA, IA, DS]) verifyTransfers(ledger driver.Ledger, transferActions []TA, signatureProvider driver.SignatureProvider, attributes driver.ValidationAttributes) error {
+func (v *Validator[P, T, TA, IA, DS]) verifyTransfers(
+	tokenRequest *driver.TokenRequest,
+	ledger driver.Ledger,
+	transferActions []TA,
+	signatureProvider driver.SignatureProvider,
+	attributes driver.ValidationAttributes,
+) error {
 	v.Logger.Debugf("check sender start...")
 	defer v.Logger.Debugf("check sender finished.")
 	for i, action := range transferActions {
-		if err := v.verifyTransfer(action, ledger, signatureProvider, attributes); err != nil {
+		if err := v.verifyTransfer(tokenRequest, action, ledger, signatureProvider, attributes); err != nil {
 			return errors.Wrapf(err, "failed to verify transfer action at [%d]", i)
 		}
 	}
 	return nil
 }
 
-func (v *Validator[P, T, TA, IA, DS]) verifyTransfer(tr TA, ledger driver.Ledger, signatureProvider driver.SignatureProvider, attributes driver.ValidationAttributes) error {
+func (v *Validator[P, T, TA, IA, DS]) verifyTransfer(
+	tokenRequest *driver.TokenRequest,
+	action TA,
+	ledger driver.Ledger,
+	signatureProvider driver.SignatureProvider,
+	attributes driver.ValidationAttributes,
+) error {
 	context := &Context[P, T, TA, IA, DS]{
 		Logger:            v.Logger,
 		PP:                v.PublicParams,
+		TokenRequest:      tokenRequest,
 		Deserializer:      v.Deserializer,
-		TransferAction:    tr,
+		TransferAction:    action,
 		Ledger:            ledger,
 		SignatureProvider: signatureProvider,
 		MetadataCounter:   map[MetadataCounterID]int{},
```

Comment:
What I don't find very neat here is that sometimes we pass a transfer action in the context, sometimes an issue action and sometimes none. I would pass none of them and then have different validator types:
```
type TransferValidator interface {
 Validate(Context, TransferAction)
}
```
Similarly for `IssueValidator` and `AuditValidator`